lib/scan_result.dart

```diff
@@ -78,12 +78,12 @@ class AdvertisementData {
   }
 
   static Uint8List _decodeBase64OrNull(String base64Value) {
-    if (base64Value != null)
+    if (base64Value != null) {
       return base64.decode(base64Value);
-    else
+    } else {
       return null;
+    }
   }
-
-  static List<String> _mapToListOfStringsOrNull(List<dynamic> values) =>
-      values?.cast();
 }
+
+List<String> _mapToListOfStringsOrNull(List<dynamic> values) => (values ?? []).cast();
```

Comment:
Why the change of behaviour? Most of those properties might be an array or null on native and that's what this function did - either map dynamics from the list to strings or preserve the null.

Reply:
Wouldn't it be a better idea to, with recent nullsafety changes in mind, return an empty list instead of null?
Otherwise it would be no problem to just continue with the `values?.cast()`

Reply:
Null is a more definitive indication that there are no such values retrieved. I like the idea of eliminating null, but I'd have to examine the rest of this file and analyze if this could be breaking or not. I'll try to do that tomorrow.